src/config.js

```diff
@@ -0,0 +1,1602 @@
+      for (let scopeSelector in selectorsAndSettings) {
+        let settings = selectorsAndSettings[scopeSelector]
+        settings = this.makeValueConformToSchema(null, settings, {suppressException: true})
+        this.setRawScopedValue(null, settings, source, scopeSelector)
+      }
+    })
+  }
+
+  /*
+  Section: Private Scoped Settings
+  */
+
+  priorityForSource (source) {
+    if (source === this.getUserConfigPath()) {
+      return 1000
+    } else {
+      return 0
+    }
+  }
+
+  emitChangeEvent () {
+    if (!(this.transactDepth > 0)) { return this.emitter.emit('did-change') }
+  }
+
+  resetUserScopedSettings (newScopedSettings) {
+    const source = this.getUserConfigPath()
+    const priority = this.priorityForSource(source)
+    this.scopedSettingsStore.removePropertiesForSource(source)
+
+    for (let scopeSelector in newScopedSettings) {
+      let settings = newScopedSettings[scopeSelector]
+      settings = this.makeValueConformToSchema(null, settings, {suppressException: true})
+      const validatedSettings = {}
+      validatedSettings[scopeSelector] = withoutEmptyObjects(settings)
+      if (validatedSettings[scopeSelector] != null) { this.scopedSettingsStore.addProperties(source, validatedSettings, {priority}) }
+    }
+
+    return this.emitChangeEvent()
+  }
+
+  setRawScopedValue (keyPath, value, source, selector, options) {
+    if (keyPath != null) {
+      const newValue = {}
+      setValueAtKeyPath(newValue, keyPath, value)
+      value = newValue
+    }
+
+    const settingsBySelector = {}
+    settingsBySelector[selector] = value
+    this.scopedSettingsStore.addProperties(source, settingsBySelector, {priority: this.priorityForSource(source)})
+    return this.emitChangeEvent()
+  }
+
+  getRawScopedValue (scopeDescriptor, keyPath, options) {
+    let legacyScopeDescriptor
+    scopeDescriptor = ScopeDescriptor.fromObject(scopeDescriptor)
+    const result = this.scopedSettingsStore.getPropertyValue(
+        scopeDescriptor.getScopeChain(),
+        keyPath,
+        options
+      )
+
+    legacyScopeDescriptor = this.getLegacyScopeDescriptor(scopeDescriptor)
+    if (result != null) {
+      return result
+    } else if (legacyScopeDescriptor) {
+      return this.scopedSettingsStore.getPropertyValue(
+          legacyScopeDescriptor.getScopeChain(),
+          keyPath,
+          options
+        )
+    }
+  }
+
+  observeScopedKeyPath (scope, keyPath, callback) {
+    callback(this.get(keyPath, {scope}))
+    return this.onDidChangeScopedKeyPath(scope, keyPath, event => callback(event.newValue))
+  }
+
+  onDidChangeScopedKeyPath (scope, keyPath, callback) {
+    let oldValue = this.get(keyPath, {scope})
+    return this.emitter.on('did-change', () => {
+      const newValue = this.get(keyPath, {scope})
+      if (!_.isEqual(oldValue, newValue)) {
+        const event = {oldValue, newValue}
+        oldValue = newValue
+        return callback(event)
```

Comment:
i don't think that event-kit's emitter actually cares about the callback return value.